ytypes/node.go

```diff
@@ -407,22 +428,17 @@ func retrieveNodeList(schema *yang.Entry, root interface{}, path, traversedPath
 
 		match := true
 		for i := 0; i < k.NumField(); i++ {
-			fieldName := listKeyT.Field(i).Name
+			fieldName := listKeyFieldNameMap[i]
 			fieldValue := k.Field(i)
 			if !fieldValue.IsValid() {
 				return nil, status.Errorf(codes.InvalidArgument, "invalid field %s in %T", fieldName, k)
 			}
 
-			elemFieldT, ok := listElemT.Elem().FieldByName(fieldName)
+			schemaKey, ok := schemaKeyMap[fieldName]
 			if !ok {
 				return nil, status.Errorf(codes.NotFound, "element struct type %v does not contain key field %s", listElemT, fieldName)
 			}
```

Comment:
If you place this error statement above, then this error can be changed to say codes.Internal, and that it is expected that this field should always be populated.

Reply:
These are good suggestions, thanks. Changes made in the latest commit.